sdk/monitor/Azure.Monitor.Ingestion/src/LogsIngestionClient.cs

```diff
@@ -360,6 +375,56 @@ public virtual async Task<Response> UploadAsync<T>(string ruleId, string streamN
             return runningTasks.Select(_ => _.CurrentTask).Last().Result; //204 - response of last batch with header
         }
 
+        /// <summary>
+        /// test
+        /// </summary>
+        /// <param name="logsCount"></param>
+        /// <param name="options"></param>
+        /// <param name="response"></param>
+        /// <param name="cancellationToken"></param>
+        protected internal virtual async void OnException(int logsCount, UploadLogsOptions options, Response response, CancellationToken cancellationToken = default)
```

Comment:
This probably belongs on the Options type rather than the client to go along with where the event is defined.